sdk/ml/azure-ai-ml/tests/test_configs/dsl_pipeline/dynamic_input_do_while/pipeline.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+
+from azure.ai.ml import Input, load_component
+from azure.ai.ml.dsl import pipeline
+from azure.ai.ml.dsl._do_while import do_while
+
+component_func = load_component(Path(__file__).parent / "component.yml")
+
+
+@pipeline
+def loop_func(**kwargs):
+    node = component_func(
+        required_input=kwargs["required_input"],
+        optional_input=kwargs["optional_input"],
+    )
+    return {
+        "output1": node.outputs.output1,
+        "output2": node.outputs.output2,
+    }
+
+
+@pipeline
+def pipeline_func(required_input: Input(type="uri_folder")):
+    init_inputs = {
+        "required_input": required_input,
+        "optional_input": None,
+    }
+    loop_body = loop_func(**init_inputs)
+    do_while(
+        body=loop_body,
```

Comment:
Add a CLI pipeline job test case?

Reply:
Can we omit input type in CLI? I though this scenario will only occur in SDK?

Reply:
oh, yes. you are right.